FinanceTracker/FinanceTracker/Utils/UserInputReader.cs

```diff
@@ -1,15 +1,42 @@
 ﻿using FinanceTracker.Enums;
+using FinanceTracker.Models;
 
 namespace FinanceTracker.Utils
 {
     public class UserInputReader
     {
+        public int GetCategory(OperationType type)
+        {
+            const string FILEPATH = "categories.json";
+            List<Category> categories = JsonFileManager.ReadCategoriesFromJson(FILEPATH);
+            categories = categories.Where(category => category.Type == type).ToList();
+
+            bool validEntry = false;
+            int categoryId = 0;
+            string userInput;
+            const string displayMessage = "Enter a category from listed below:";
+
+            while (!validEntry)
+            {
+                userInput = GetInput(displayMessage, categories).ToLower();
+                foreach (var category in categories)
+                {
+                    if (userInput == category.Name?.ToLower())
+                    {
+                        categoryId = category.Id;
+                        validEntry = true;
```

Comment:
Maybe you should break after this line? Also, instead this loop you can use .FirstOrDefault from LINQ